server/chroma/src/main.rs

```diff
@@ -1,52 +1,98 @@
 extern crate core;
 
-use crate::config::Config;
-use crate::routes::appdata::{AlbumIdCache, AppData, Ratelimits, SessionIdCache, WebData};
-use crate::routes::routable::Routable;
+use std::process;
+use std::time::Duration;
+
 use actix_cors::Cors;
 use actix_web::{web, App, HttpServer};
 use cabbage::KoalaApi;
-use color_eyre::eyre::Error;
 use color_eyre::Result;
-use dal::database::Database;
-use dal::storage_engine::{S3Config, Storage};
 use dotenv::dotenv;
 use noiseless_tracing_actix_web::NoiselessRootSpanBuilder;
-use std::time::Duration;
-use tracing::{info, warn};
+use tracing::{error, info, warn};
 use tracing_actix_web::TracingLogger;
 use tracing_subscriber::fmt::layer;
 use tracing_subscriber::layer::SubscriberExt;
 use tracing_subscriber::util::SubscriberInitExt;
 use tracing_subscriber::EnvFilter;
 
+use dal::database::Database;
+use dal::storage_engine::{S3Config, Storage};
+
+use crate::config::Config;
+use crate::routes::appdata::{AlbumIdCache, AppData, Ratelimits, SessionIdCache, WebData};
+use crate::routes::routable::Routable;
+
 mod config;
 mod routes;
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    dotenv().ok();
+    // Try to load the environment variables from the .env file
+    if let Err(err) = dotenv() {
+        panic!("failed to load .env file: {:#}", err);
```

Comment:
Won't that print a whole backtrace (which isn't needed for an IO-problem as a corrupted or locked .env file)?

Reply:
It can, if enabled with RUST_BACKTRACE. But the backtrace here is a short one, some rustc stuff, some tokio stuff, and our main.

Reply:
Is there an added benefit to using `color_eyre` then?

Reply:
Certainly, if a fatal error occurs deeper in the call stack it can be very useful. Furthermore, at startup I don't particularly care about the error type, and color_eyre can conver from any std::error::Error, I just care that there is an error and want to bubble it up.

So why not panic? Few reasons:
- Drop destructors aren't always properly run
- Not as nice for the end user, compared to color_eyre
- If panic=abort is set, which I quite often do, you get nothing, the application just halts

Reply:
I agree with not using panic, however, I am not entirely convinced of using color_eyre. Having a stack trace is not nice for the end user  as a stack trace indicates a fundamental mistake in the program, not a problem with the system. Having a problem occur with dotenv() almost certainly indicates a problem with the system (missing file, readlock, etc.), not a problem with the rust code. In my opinion, producing a stack trace here gives the end user the wrong idea (a problem that they can't fix and should be fixed by the developer). Therefore, I propose to change the code to match the style of the rest of the error messages:
```rust
dotenv().unwrap_or_else(|err| {
    error!("failed to load .env file: {:#}", err);
    process::exit(1);
});
```